scopesim_templates/stellar/stars.py

```diff
@@ -243,6 +252,10 @@ def stars(filter_name, amplitudes, spec_types, x, y, library="pyckles", ra=RA0,
     ref_dict = {spt: ii for ii, spt in enumerate(unique_types)}
     ref = np.array([ref_dict[i] for i in spec_types])
 
+    if len(ref) == 1:
+        ref = ref[0] * np.ones(len(x))
+        spec_types *= len(x)
```

Comment:
This seems broken to me, because ref is an array of integers at the start, say
```
len_x = 10
ref = np.array([0])
```
then
```
ref = ref[0] * np.ones(len_x)
```
leads to an array of floats:
```
array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
```
These floats are subsequently used in `FieldOfView.extract_from(self, src)` in `fov.py` in ScopeSIM:
```
        spectra = {ref: fu.extract_range_from_spectrum(src.spectra[ref], waves)
                   for ref in np.unique(spec_refs)}
```
And this fails with
```
TypeError: list indices must be integers or slices, not numpy.float64
```
This also means that there is no comprehensive test that tests a field with stars.

Reply:
Thanks for picking up that hole with the Source object.

I've added an [issue to ScopeSim](https://github.com/AstarVienna/ScopeSim/issues/142) to fix it